### PR TITLE
🔀 Enhance upload asset page and connect API

### PR DESCRIPTION
**Pull request by mileswong**
connects #25

### CODE REVIEW

**Comment by rickmak** [comment]
hi @mileswong May you take a look at the PR and rebase to resolve the conflict and travis fails?

File: util/fileUtil.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    return window.crypto.subtle.digest('SHA-256', buf).then(this.buf2hex);
+  },
+  blobToArrayBuffer(blob) {
+    return new Promise((resolve) => {
+      const reader = new FileReader();
+      reader.onload = () => resolve(reader.result);
+      reader.readAsArrayBuffer(blob);
+    });
+  },
+  buf2hex(buffer) {
```
Comment (nwingt) [inline util/fileUtil.js]: For consistent naming, please change the function name to `arrayBufferToHex`

File: util/fileUtil.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    // buffer is an ArrayBuffer
+    return Array.prototype.map
+      .call(new Uint8Array(buffer), (x) => `00${x.toString(16)}`.slice(-2))
+      .join('');
+  },
+};
+
+export default fileUtil;
```
Comment (nwingt) [inline util/fileUtil.js]: How about exporting each function instead of exporting all functions as an object?

File: store/modules/actions/user.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+import axios from '@/plugins/axios';
+
+import * as types from '@/store/mutation-types';
+import ethUtil from '@/util/ethUtil';
+
+async function checkIsUser(wallet) {
+  try {
+    return await axios.get(`api/users/wallet/${wallet}`);
```
Comment (nwingt) [inline store/modules/actions/user.js]: `/api`

File: store/modules/actions/user.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  commit(types.USER_SET_USER_INFO, userInfo);
+}
+
+export async function onWalletChanged(ctx, wallet) {
```
Comment (nwingt) [inline store/modules/actions/user.js]: rename as `updateWallet`

File: constant/index.js [hunk 5]
```diff
@@ -20,4 +22,11 @@ export const SUPPORTED_LICENSE = new Set([
   'cc-by-nd',
 ]);
 
+export const LICENSE_URL = {
+  'cc-by': 'https://creativecommons.org/licenses/by/4.0/',
+  'cc-by-nd': 'https://creativecommons.org/licenses/by-nd/4.0/',
+  'cc-by-sa': 'https://creativecommons.org/licenses/by-sa/4.0/',
+  cc0: 'https://creativecommons.org/share-your-work/public-domain/cc0/',
+};
+
```
Comment (nwingt) [inline constant/index.js]: How about merging with `SUPPORTED_LICENSE` as:
```javascript
export const LICENSE = {
  'cc-by': 'https://creativecommons.org/licenses/by/4.0/',
  'cc-by-nd': 'https://creativecommons.org/licenses/by-nd/4.0/',
  'cc-by-sa': 'https://creativecommons.org/licenses/by-sa/4.0/',
  cc0: 'https://creativecommons.org/share-your-work/public-domain/cc0/',
};

export const SUPPORTED_LICENSE = new Set(Object.keys(LICENSE));
```

**Comment by rickmak** [comment]
@mileswong please avoid merge master back to feature branch at 33b6eae.

This project is using rebase workflow.

**Pull request merged**